session.go

```diff
@@ -460,6 +472,13 @@ func ParseURL(url string) (*DialInfo, error) {
 		MinPoolSize:    minPoolSize,
 		MaxIdleTimeMS:  maxIdleTimeMS,
 	}
+	if ssl && info.DialServer == nil {
```

Comment:
@domodwyer I added `&& info.DialServer == nil` here, to ensure we don't override user's settings.